src/cart/services/cart.service.ts

```diff
@@ -1,31 +1,60 @@
 import { Injectable } from '@nestjs/common';
-
 import { v4 } from 'uuid';
 
 import { Cart } from '../models';
+import { DbClientService } from '../../shared/services';
 
 @Injectable()
 export class CartService {
-  private userCarts: Record<string, Cart> = {};
+  constructor(
+    private dbClientService: DbClientService
+  ) {
+  }
 
-  findByUserId(userId: string): Cart {
-    return this.userCarts[ userId ];
+  async findByUserId(userId: string): Promise<Cart> {
+    const client = await this.dbClientService.getClient();
+    const query = await client.query(`
+      SELECT carts.id, user_id, product_id, count, title, description, price FROM carts
```

Comment:
Please reconsider your approach to commands creation. Currently your code is vulnerable to SQL Injection attack.
Please use parameterized commands:
https://node-postgres.com/features/queries#parameterized-query